### DISCUSSION TITLE
Silicon Labs Gecko SDK 5.0.0.0 and Silabs Simplicity SDK Suite 2024.6.0 been announced and to be released soon

### DISCUSSION THREAD

**Discussion by Hedda** [darkxst/silabs-firmware-builder#100] [General]
FYI, Silicon Labs Gecko SDK 5.0.0.0 and Silabs Simplicity SDK Suite 2024.6.0 have been announced but not yet released(?).

Perhaps more noteworthy is that Silicon Labs is phasing older SoC generations by not including them in newer SDK versions, so while long-term support for those will continue with bug-fixes only for previous SDK versions, only the new SDKs will get new features/functions.

https://www.silabs.com/developers/software-development-kit-release-notes

https://www.silabs.com/developers/simplicity-software-development-kit

Gecko SDK 5.0 seems to be scheduled to be released very soon as Q2/2024 (June 2024) is specifically mentioned as the release date:

https://www.silabs.com/documents/public/release-notes/gecko-platform-release-notes-5.0.0.0.pdf

https://github.com/SiliconLabs/gecko_sdk

https://github.com/SiliconLabs/simplicity_sdk

Note! EmberZNet 8.0.0.0 and Gecko Platform remove support for EFR32 Series 0/1 (e.i. EFR32MGx and EFR32MG1x) so need to use separate SDKs for Series 0 & 1 and Series 2 & 3 Starting in Q2 2024 (June 2024) as the new Gecko Platfoprm and SDK are designed specifically for our EFR32 Series 2 and upcoming Series 3 devices only.

"_Meanwhile, our Gecko Software Development Kit (GSDK) will continue to be available for users of our Series 0 and Series 1 devices. The Gecko SDK will continue to support all [Series 0 and 1 devices](https://www.silabs.com/developers/software-development-kit-release-notes/series-directory) with no change to the long-term support, maintenance, quality, and responsiveness provided under our software policy.The new SDK will branch from Gecko SDK and begin to offer new features that help developers take advantage of the advanced capabilities of our Series 2 and 3 products._"

### New in Gecko Platform Version 5.0.0.0
* Migration to Simplicity SDK Suite
* Removed support for Series 0/1
 * Peripherals
  * New API names introduced for low-level drivers (with compatibility layer for old names)
* Services
  * Memory Manager, APIs to manage embedded dynamic memory allocations
  * Clock Manager, for clock tree initialization
  * Event System, for inter-process communication
  * Interrupt manager, APIs to manage embedded interrupts
  * APIs for accessing value of tokens added to Token Manager
* CPC
  * Small improvement and bug fixes
* Security
  * Mbed TLS upgraded (to version 3.5.0)
* RAIL 5.0.0
  * Added support for collision detection feature on the EFR32xG25 parts
  * Added support for additional Coex TX and RX metrics events to RAIL coexistence utility
  * Updated some RAIL APIs to better prepare for future API changes while retaining backwards compile-time compatibility
* Other Components
  * Compilers upgraded (to GCC 12.2.1 and IAR 9.40.1)

![image](https://github.com/zigpy/zigpy/assets/6320001/24124a98-49d1-448f-9dfc-19ae5a6fb58a)

Also new is OpenThread version 2.5.0.0, Bluetooth 8.0.0, and Z-Wave Long Range 800 version 7.22.0.0 + USB Device Stack 1.3.0.0

EmberZNet 8.0.0.0 brings support for EFR32MG26 (and MGM260) also known as xG26 as well:

https://www.silabs.com/documents/public/release-notes/emberznet-release-notes-8.0.0.0.pdf

#### New Platform Support in release EmberZNet 8.0.0.0
•  New hardware xG26 and xG22E.
•  new platform component, Clock Manager, is integrated into all Zigbee sample applications for clock configuration.

#### New Components in release 8.0.0.0
• Inter Process Communication
Added a new component named “zigbee_ipc”.
This component adds support for interprocess communication on a Zigbee RTOS device.
• Byte utilities
Added a new utility component named “byte_utilities” outside Zigbee protocol for data store and fetch routines.
As part of that, moved the header file byte-utilities.h from protocol/zigbee/stack/include/ to util/plugin/byte_utilities.
Since this is not specific to Zigbee, the API has been renamed, from sl_zigbee_reverse_mem_copy() to sl_util_reverse_mem_copy().
• Separated a new component "Zigbee System Common (App Framework Task)" from Zigbee System Common, to better allow for apps that do not bring in the full ZCL and app framework but still use the app framework's RTOS task.
• RTOS Configurations

#### New APIs in release 8.0.0.0
• A new event initialization function sl_zigbee_af_isr_event_init is added for the events that can be activated in ISR.
Note that with this release the APIs have been renamed. For details, see the Zigbee v8.0.0 section of https://docs.silabs.com/.

#### New Application in release 8.0.0.0
• Zigbee Unified Test Harness SoC application is introduced as a new application in this release.
• Added a new ZigbeeMinimalRtos app to Zigbee offerings. This is a test application with a bare minimum RTOS based Zigbee stack. This application is non-compliant since it does not have all the necessary component and configurations of a fully functional Zigbee application.
• A new configuration for the SPI based NCP as a sleepy NCP is introduced in this release. This serves the use case of a host-SPI NCP configuration to be used as a sleepy node in the network. The host controls the sleep and wake up of the SPI NCP. See AN711: SPI Host Interfacing Guide for Zigbee for more details.

#### New Platform Support in release 8.0.0.0
•  New hardware xG26 and xG22E.
•  new platform component, Clock Manager, is integrated into all Zigbee sample applications for clock configuration.

**Comment by Hedda** [comment]
@darkxst FYI, latest ZHA/zigpy and latest ember driver for Zigbee2MQTT (Z2M) now has experimental support for EmberZNet 8.0.1.0 See:

* https://github.com/Koenkk/zigbee2mqtt/discussions/23224
  * https://github.com/Koenkk/zigbee-herdsman/pull/1094
    * https://github.com/Koenkk/zigbee-herdsman/issues/1093

and

* https://github.com/zigpy/bellows/pull/631

Nerivec provides some experimental EmberZNet 8.0.0.0 firmware builds built from his own fork of silabs-firmware-builder:

* https://github.com/Nerivec/silabs-firmware-builder/releases/tag/v8.0.0-pre

* https://github.com/Nerivec/silabs-firmware-builder/releases/tag/v2024.6.1

puddly also built some EmberZNet 8.0.0.0 ffirmware for Home Assistant SkyConnect / ZBT-1 (and Yellow):

* https://github.com/NabuCasa/silabs-firmware-builder/actions/runs/9614847084

PS: Nerivec suggests using his ember-zli for flashing (which in turn depends on the latest zigbee-herdsman) with comment to report issues:

* https://github.com/Nerivec/ember-zli 

* https://github.com/Nerivec/ember-zli/issues

**Comment by darkxst** [comment]
I have v8 builds for a while in this branch, need to do some testing then will merge into main:

https://github.com/darkxst/silabs-firmware-builder/tree/2024.06
